src/select.rs

```diff
@@ -93,21 +95,28 @@ impl<'a> TElement<'a> for &'a ElementData {
 }
 
 impl<'a> Node<'a> {
-    pub fn select(&'a self, css_str: &str) -> Result<FilterNodes<Descendants<'a>>, ()> {
+    pub fn select(&'a self, css_str: &str) -> Result<FilterNodes<Descendants<'a>, Vec<Selector>>, ()> {
         let selectors = try!(parser::parse_author_origin_selector_list_from_str(css_str));
         Ok(FilterNodes{
             iter: self.descendants(),
             filter: selectors,
         })
     }
+
+    pub fn text_iter(&'a self) -> FilterNodes<Siblings<'a>, TextFilter> {
+        FilterNodes {
+            iter: self.children(),
```

Comment:
Is taking the text on only direct child text nodes (ignoring other descendants) ever the desired behavior?